modVM/src/org/aion/vm/AbstractExecutionResult.java

```diff
@@ -119,79 +147,65 @@ public long getNrgLeft() {
     }
 
     /**
-     * Sets nrg left.
+     * Returns the AbstractExecutionResult's output.
+     *
+     * @return the output.
      */
-    public void setNrgLeft(long nrgLeft) {
-        this.nrgLeft = nrgLeft;
+    public byte[] getOutput() {
+        return output;
     }
 
     /**
-     * Sets this ContractExecutionResult's code to code and its energy left to nrgLeft.
+     * Sets the AbstractExecutionResult's code to the code whose integer representation is code.
      *
-     * @param code The new code.
-     * @param nrgLeft The new nrgLeft.
+     * @param code the integer representation of the code to be set.
      */
-    public void setCodeAndNrgLeft(int code, long nrgLeft) {
-        setCode(code);
+    public void setCode(int code) {
+        ResultCode resCode = ResultCode.fromInt(code);
+        this.code = resCode;
```

Comment:
the resCode variable is not actually needed

Reply:
thanks, I had intended to check for null first